File: app/src/main/java/ga/forntoh/bableschool/CategoryActivity.java
package ga.forntoh.bableschool;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.view.Menu;
import android.view.MenuInflater;
import android.widget.TextView;

import com.google.gson.Gson;

import ga.forntoh.bableschool.fragments.CourseNotesFragment;
import ga.forntoh.bableschool.fragments.ForumFragment;
import ga.forntoh.bableschool.fragments.MyProfileFragment;
import ga.forntoh.bableschool.fragments.NewsFragment;
import ga.forntoh.bableschool.fragments.RankingFragment;
import ga.forntoh.bableschool.model.Category;

public class CategoryActivity extends BaseActivity {

    TextView toolbar_title;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_category);
        disableFlags(true);

        Category category = new Gson().fromJson(getIntent().getStringExtra("category"), Category.class);
        toolbar_title = findViewById(R.id.title);
        toolbar_title.setText(category.getTitle());

        setSupportActionBar(findViewById(R.id.toolbar));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        enableWhiteStatusBar();

        switch (category.getId()) {
            case 0:
                loadFragment(new NewsFragment());
                break;
            case 1:
                loadFragment(new CourseNotesFragment());
                break;
            case 2:
                loadFragment(new RankingFragment());
                break;
            case 3:
                loadFragment(new ForumFragment());
                break;
            case 4:
                loadFragment(new MyProfileFragment());
            default:
                break;
        }
    }

    public FragmentTransaction loadFragment(Fragment fragment) {
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.frame_container, fragment);
        transaction.commit();
        return transaction;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_home, menu);
        return true;
    }
}

File: app/src/main/java/ga/forntoh/bableschool/CategoryActivity.kt
package ga.forntoh.bableschool

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentTransaction
import android.view.Menu
import android.widget.TextView
import com.google.gson.Gson
import ga.forntoh.bableschool.fragments.*
import ga.forntoh.bableschool.model.Category

class CategoryActivity : BaseActivity() {

    private val toolbarTitle by lazy { findViewById<TextView>(R.id.title) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_category)
        disableFlags(true)

        val (id, title) = Gson().fromJson(intent.getStringExtra("category"), Category::class.java)
        toolbarTitle.text = title

        setSupportActionBar(findViewById(R.id.toolbar))
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        enableWhiteStatusBar()

        when (id.toInt()) {
            1 -> loadFragment(NewsFragment())
            2 -> loadFragment(CourseNotesFragment())
            3 -> loadFragment(RankingFragment())
            4 -> loadFragment(ForumFragment())
            5 -> loadFragment(MyProfileFragment())
        }
    }

    fun loadFragment(fragment: Fragment): FragmentTransaction =
            supportFragmentManager.beginTransaction().apply {
                replace(R.id.frame_container, fragment)
                commit()
            }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu_home, menu)
        return true
    }
}
